From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gerson Fernando Budke <gerson.budke@ossystems.com.br>
Subject: [PATCH] STM32_ADC_Common: Fix watchdog 2/3 bits

The watchdog 2 and 3 can selected by the channel bit. This reserves the
bits to match with the number of channels.

Signed-off-by: Gerson Fernando Budke <gerson.budke@ossystems.com.br>

diff --git a/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs b/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs
--- a/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs
+++ b/src/Emulator/Peripherals/Peripherals/Analog/STM32_ADC_Common.cs
@@ @@ private Dictionary<long, DoubleWordRegister> BuildRegistersMap(bool hasCalibrati
                     .WithValueField(16, 12, out analogWatchdogHighValues[1], name: "HT")
                     .WithReservedBits(28, 4));
                 registers.Add((long)Registers.Watchdog2Configuration, new DoubleWordRegister(this)
-                    .WithFlags(0, 14, out analogWatchdog2SelectedChannels, name: "AWD2CH")
-                    .WithReservedBits(14, 18));
+                    .WithFlags(0, ChannelCount, out analogWatchdog2SelectedChannels, name: "AWD2CH")
+                    .WithReservedBits(ChannelCount, 32 - ChannelCount));
             }
             if(WatchdogCount == 3)
             {
@@ @@ private Dictionary<long, DoubleWordRegister> BuildRegistersMap(bool hasCalibrati
                     .WithValueField(16, 12, out analogWatchdogHighValues[2], name: "HT")
                     .WithReservedBits(28, 4));
                 registers.Add((long)Registers.Watchdog3Configuration, new DoubleWordRegister(this)
-                    .WithFlags(0, 14, out analogWatchdog3SelectedChannels, name: "AWD3CH")
-                    .WithReservedBits(14, 18));
+                    .WithFlags(0, ChannelCount, out analogWatchdog3SelectedChannels, name: "AWD3CH")
+                    .WithReservedBits(ChannelCount, 32 - ChannelCount));
             }
 
             if(hasCalibration)
